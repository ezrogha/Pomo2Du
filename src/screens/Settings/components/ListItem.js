import React from 'react'
import { Text, View, TouchableOpacity, StyleSheet } from 'react-native'

export default ({ title, value }) => (
    <TouchableOpacity style={styles.listItem}>
        <View>
            <Text style={styles.listItemText}>{title}</Text>
        </View>
        <View>
            <Text style={styles.listItemText}>{value}</Text>
        </View>
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    listItem: {
        height: 50,
        backgroundColor: "#fff",
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center",
        paddingRight: 10,
        borderBottomColor: 'lightgrey',
        borderBottomWidth: 0.4
    },
    listItemText: {
        fontSize: 16,
        color: "rgb(100, 100, 100)"
    },
})
