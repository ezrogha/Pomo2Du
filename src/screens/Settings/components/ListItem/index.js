import React from 'react'
import { Text, View, TouchableOpacity, StyleSheet } from 'react-native'

export default ({ title, value, triggerModal, chooseAlarm }) => (
    <TouchableOpacity style={styles.listItem} onPress={triggerModal || chooseAlarm}>
        <View>
            <Text style={[styles.listItemText, { color: "#333" } ]}>{title}</Text>
        </View>
        <View>
            <Text style={[styles.listItemText, { color: "#aaa" }]}>{value}</Text>
        </View>
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    listItem: {
        height: 50,
        backgroundColor: "#fff",
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center",
        paddingRight: 10,
        borderBottomColor: 'lightgrey',
        borderBottomWidth: 0.4
    },
    listItemText: {
        fontSize: 16,
        color: "rgb(100, 100, 100)"
    },
})
