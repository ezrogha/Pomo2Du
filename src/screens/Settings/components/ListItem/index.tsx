import React from 'react'
import { Text, View, TouchableOpacity, StyleSheet } from 'react-native'

type ModalProps = {
    list: any[],
    selectedValue: number,
    listItem: string,
    metric: string
}

interface Props {
    title: string,
    value: string,
    triggerModal?: () => void,
    chooseAlarm?: (route: string) => void
}

const ListItem: React.FC<Props> = ({ title, value, triggerModal, chooseAlarm }) => (
    <TouchableOpacity style={styles.listItem} onPress={triggerModal || chooseAlarm}>
        <View>
            <Text style={[styles.listItemText, { color: "#333" } ]}>{title}</Text>
        </View>
        <View>
            <Text style={[styles.listItemText, { color: "#aaa" }]}>{value}</Text>
        </View>
    </TouchableOpacity>
)

const styles = StyleSheet.create({
    listItem: {
        height: 50,
        backgroundColor: "#fff",
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "center",
        paddingRight: 10,
        borderBottomColor: 'lightgrey',
        borderBottomWidth: 0.4
    },
    listItemText: {
        fontSize: 16,
        color: "rgb(100, 100, 100)"
    },
})

export default ListItem;
