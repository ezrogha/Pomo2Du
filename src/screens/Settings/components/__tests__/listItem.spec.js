import React from 'react'

import ListItem from '../ListItem';
import renderer from 'react-test-renderer'

describe("Testing ListItem", () => {
    test("renders correctly", () => {
        const tree = renderer.create(<ListItem />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
