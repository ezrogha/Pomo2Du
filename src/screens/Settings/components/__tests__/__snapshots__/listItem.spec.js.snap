// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ListItem should render correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderBottomColor": "lightgrey",
      "borderBottomWidth": 0.4,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingRight": 10,
    }
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(100, 100, 100)",
            "fontSize": 16,
          },
          Object {
            "color": "#333",
          },
        ]
      }
    />
  </View>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(100, 100, 100)",
            "fontSize": 16,
          },
          Object {
            "color": "#aaa",
          },
        ]
      }
    />
  </View>
</View>
`;
